src/theory/arith/proof_checker.cpp

```diff
@@ -407,6 +409,72 @@ Node ArithProofRuleChecker::checkInternal(ProofRule id,
       }
       return args[0];
     }
+    case ProofRule::ARITH_POLY_NORM_REL:
+    {
+      Assert(children.size() == 1);
+      Assert(args.size() == 1);
+      Kind k;
+      if (!getKind(args[0], k)
+          || (k != Kind::LT && k != Kind::LEQ && k != Kind::EQUAL
+              && k != Kind::GT && k != Kind::GEQ))
+      {
+        return Node::null();
+      }
+      if (children[0].getKind() != Kind::EQUAL)
+      {
+        return Node::null();
+      }
+      Node l = children[0][0];
+      Node r = children[0][1];
+      if (l.getKind() != Kind::MULT || r.getKind() != Kind::MULT)
+      {
+        return Node::null();
+      }
+      Node lr = l[1];
+      Node rr = r[1];
+      if (lr.getKind() != Kind::SUB || rr.getKind() != Kind::SUB)
+      {
+        return Node::null();
+      }
+      Node cx = l[0];
+      Node x1 = lr[0];
+      Node x2 = lr[1];
+      Node cy = r[0];
+      Node y1 = rr[0];
+      Node y2 = rr[1];
+      if (cx.getKind() == Kind::CONST_RATIONAL
```

Comment:
Can you merge this block with the one below by checking that cx and cy are constants of real or int type?